server/src/main/java/org/apache/druid/server/StatusResource.java

```diff
@@ -92,15 +96,15 @@ private Map<String, String> filterHiddenProperties(
       Map<String, String> allProperties
   )
   {
-    return Maps.filterEntries(
-        allProperties,
-        (entry) -> hiddenProperties
-            .stream()
-            .anyMatch(
-                hiddenPropertyElement ->
-                    !StringUtils.toLowerCase(entry.getKey()).contains(StringUtils.toLowerCase(hiddenPropertyElement))
-            )
+    Map<String, String> propertyCopy = new HashMap<>(allProperties);
+    allProperties.keySet().forEach(
```

Comment:
The set of hidden properties is likely to be much smaller than the set of all properties. Should we iterate over the hidden properties so we have a much lower total cost? One could argue that this operation is used infrequently and efficiency doesn't matter. But, since we're changing the code anyway, we might as well make it simple.

Reply:
Isn't this going to be O(N*M) regardless since for every hidden property we would have to check every property and vice versa? In fact, for the current implementation we sometimes get a short-circuit on the nested iteration since it is using `anyMatch`.